bootstrap/spigot/src/main/java/org/geysermc/geyser/platform/spigot/world/manager/GeyserSpigotWorldManager.java

```diff
@@ -128,16 +130,15 @@ public void sendLecternData(GeyserSession session, int x, int z, List<BlockEntit
         }
     }
 
-    private Chunk getChunk(World world, int x, int z) {
+    private @Nullable Chunk getChunk(World world, int x, int z) {
         if (!world.isChunkLoaded(x, z)) {
             return null;
         }
         return world.getChunkAt(x, z);
     }
 
     private void sendLecternData(GeyserSession session, Chunk chunk, List<BlockEntityInfo> blockEntityInfos) {
-        for (int i = 0; i < blockEntityInfos.size(); i++) {
```

Comment:
Stuff like this has been intentional. The `foreach` loop uses the iterator() method which does create another object.

Reply:
Just a heads up that there are other locations that have been replaced with a for-each loop that I didn't link on discord

Reply:
https://github.com/GeyserMC/Geyser/pull/4238/commits/8f3f82906ea697619089da5bae0e140d6a515532
This should undo those changes, i hope i didnt miss any